### PR TITLE
feat: add scratch arduino extension

### PR DESCRIPTION
**Pull request by lucarin91**
Open sourcing the Arduino scratch 3.0 extension published on https://labs-scratch.arduino.cc/ in addition to an early implementation of a specific extension for the include-robot.

### CODE REVIEW

File: scratch-arduino-extensions/scripts/patch-gui.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const path = require('path');
+const fs = require('fs');
+
+const ExtIds = ["IncludeRobot", "Scratch3Arduino"];
+const ExtDirName = "scratch3_arduino";
+const ExtDirPath = path.resolve("./scratch-vm/src/extensions/", ExtDirName);
+
+const GuiDir = "../scratch-gui";
```
Comment (umbynos) [inline scratch-arduino-extensions/scripts/patch-gui.js:8]:
```suggestion
const GuiDir = "./scratch-gui";
```

Comment (csarnataro) [reply scratch-arduino-extensions/scripts/patch-gui.js:8]: This fix is not needed, when I suggested it, my folder structure was not following the right structure. Sorry for the noise.

**Comment by csarnataro** [comment]
I would add an entry in the `.gitignore` file for the `scratch-gui` folder, so that there are less risks to add it by mistake.

```
.DS_Store
scratch-gui/
```

**Review by csarnataro — APPROVED** [review]
I left a comment about an entry in the `.gitignore` file, but apart from that, everything looks good to me.

File: scratch-arduino-extensions/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+```
+cd ..
+git clone https://github.com/scratchfoundation/scratch-gui.git
+```
+
+2. Install all dependency 
+
+```
+cd scatch-gui
```
Comment (csarnataro) [inline scratch-arduino-extensions/README.md:24]:
```suggestion
cd scratch-gui
```

File: scratch-arduino-extensions/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- [ ] Auto reload doesn't work in linux
+- [ ] Create a deployment script/ci job
+
+## Development guide
+
+1. Clone the scratch-gui folder in the top folder
+
+```
+cd ..
+git clone https://github.com/scratchfoundation/scratch-gui.git
```
Comment (dido18) [inline scratch-arduino-extensions/README.md:18]:
```suggestion
git clone --depth 1 https://github.com/scratchfoundation/scratch-gui.git
```

File: scratch-arduino-extensions/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+3. Run the patch/link script
+
+```
+node ../scratch-arduino-extensions/scripts/patch-gui.js
+```
+
+4. Start the scratch gui
+```
+npm start
+```
```
Comment (dido18) [inline scratch-arduino-extensions/README.md:37]: Add last step
```
5. Open the scratch GUI at `http://127.0.0.1:8601`
```

Comment (lucarin91) [reply scratch-arduino-extensions/README.md:37]: I agree, can you push a commit for me?

**Pull request merged**